pydatastructs/graphs/adjacency_list.py

```diff
@@ -20,7 +20,7 @@ def __new__(cls, *vertices):
         for vertex in vertices:
             obj.__setattr__(vertex.name, vertex)
         obj.vertices = [vertex.name for vertex in vertices]
-        obj.edge_weights = dict()
+        obj.edge_weights = {}
```

Comment:
The modification is significantly faster than the original version. However, this introduces ambiguity of confusing `{}` with `set`. Please add a comment after this line that it is a `dict`. See the suggestion in the below comment.

Reply:
```suggestion
        obj.edge_weights = {} # dict
```

Reply:
It's good to have comments but python by default treat `{}` as dictionary, it doesn't seem confusing.

Reply:
What is needed to be done as of now? 

> Please add a comment after this line that it is a `dict`.

This will be quite cumbersome.

> It's good to have comments but python by default treat `{}` as a dictionary, it doesn't seem confusing.

I agree^

Reply:
Adding one extra word won't do much harm but taking care of this in future may be problematic. Let's leave it as it is (without comments).